src/graph/graphQLClient.ts

```diff
@@ -17,32 +17,62 @@ import config from '../config';
  * gives a graphQLClient for pocket-graph url
  *
  * This client initializes a `graphql-request` client
- * @param access_token accessToken of the user
- * @param consumer_key consumerKey assocuated with the user
+ * @param headers any headers received by proxy is just pass through to web graphQL proxy.
+ * @param accessToken accessToken of the user
+ * @param consumerKey consumerKey associated with the user
  */
-export function getClient(accessToken: string, consumerKey: string) {
-  return new GraphQLClient(
-    `${config.graphQLProxy}?consumer_key=${consumerKey}&access_token=${accessToken}`,
-    {
+export function getClient(
+  accessToken: string,
+  consumerKey: string,
+  headers: any
+) {
+  //these headers are not compatible with GraphQLClient's fetch.
+  //they throw an error instead, so ignoring them
+  // headers might include cookie, some observability traceIds, apollo studio headers
+  // so, we are implicitly passing them and removing only those that causes issues
+  delete headers['content-length'];
```

Comment:
including these headers gives an error for `POST: v3/get: TypeError: fetch failed`. not able to identify the root cause of this yet, so excluding them

if we have an explicit list of headers that we can expect from web-repo client (e.g cookie, trace-id, and apollo related headers), we can add them explicitly to header list

Reply:
e.g error is here: https://pocket.sentry.io/issues/4034111084/?project=4504658424102912&query=is%3Aunresolved&referrer=issue-stream&stream_index=0